src/application.coffee

```diff
@@ -644,7 +653,28 @@ application.update = update = (force) ->
 			# Set the updating as finished in its own block, so it never has to worry about other code stopping this.
 			updateStatus.state = UPDATE_IDLE
 
+listenToEvents = ->
+	docker.getEventsAsync()
+	.then (stream) ->
+		parser = JSONStream.parse()
+		stream.pipe(parser)
+		parser.on 'data', (data) ->
+			if data?.Type == 'container'
+				knex('app').select().where({ containerId: data.id })
+				.then ([ app ]) ->
+					if app?
+						if data.status == 'die'
+							logSystemEvent(appExit, app)
+						else if data.status == 'restart'
+							logSystemEvent(appRestart, app)
+							logger.attach(app)
```

Comment:
It's a bit complicated, cause the event might get fired before we save `containerId` to the  DB. I'd rather make this change as unobtrusive as possible and in any case leave that for a later refactoring PR.